.github/workflows/pypi_publish.yml

```diff
@@ -0,0 +1,97 @@
+name: PyPI publish
+
+on:
+  workflow_call:
+    inputs:
+      package_version:
+        description: "The version used to publish to the registry."
+        type: string
+        required: true
+      test_pypi:
+        description: "Parameter to publish the package to Test PyPI when set to 'true', and to publish to official Pypi when set to 'false'."
+        type: string
+        required: true
+      python_latest:
+        description: 'Use the latest Python version (3.12)'
+        required: true
+        default: false
+        type: boolean
+
+jobs:
+
+  build-package:
+    name: Build Package with Verified Version
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository 
+        uses: actions/checkout@v4
+
+      - name: Validate Version Tag
+        uses: ghga-de/gh-action-pypi/verify-version-tag@main
+        with:
+          tag: ${{ inputs.package_version }}
+
+      - name: Build Python Package
+        uses: ghga-de/gh-action-pypi/build-python-package@main
+
+      - name: Upload package artifact
+        uses: actions/upload-artifact@v4
+        with:
+          path: ./dist
+
+  define-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      versions: ${{ steps.version.outputs.python }}
+    steps:
+      - name: Define python versions
+        id: version
+        run: |
+          PYTHON_LATEST=${{ inputs.python_latest }}
```

Comment:
Instead of using `python_latest` boolean as input, you could already pass a string (JSONified array) `supported_python_versions` that can be parsed with fromJSON as you do below in the test-package step. This would be more flexible and you don't need the extra step "define-matrix".

Or you can do this as part of the next PR where you derive the supported versions from the pyproject.toml file.